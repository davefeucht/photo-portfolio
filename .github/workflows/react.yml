# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
