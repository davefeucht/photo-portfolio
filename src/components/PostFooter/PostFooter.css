.post-footer {
  background-color: #000000;
  font-size: min(3vw, 1rem);
  color: #FFFFFF;
  grid-area: footer;
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  transition: width var(--modal-transition-time), height var(--modal-transition-time)
}

.post-footer .labels {
  margin: auto auto auto 5px;
}