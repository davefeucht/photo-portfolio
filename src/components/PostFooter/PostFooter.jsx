/****************
* PostTitlebar component displays the titlebar for a Post
****************/

import React from "react";
import { observer } from 'mobx-react';
import CloseButton from '../CloseButton/CloseButton.jsx';
import './PostFooter.css';

const PostFooter = observer(({ tags }) => {
  return(
    <div className="post-footer">
      <div className="labels">{tags.join(', ')}</div>
    </div>
  )
});

PostFooter.displayName = 'PostFooter';

export default PostFooter;