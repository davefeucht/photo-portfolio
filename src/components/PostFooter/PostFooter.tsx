/** **************
* PostFooter component displays the footer for a Post
*************** */

import './PostFooter.css';

import { observer } from 'mobx-react';
import * as React from 'react';

import { Store } from '../../utils/types';

interface PostFooterProps {
    stateStore: Store
}

const PostFooter: React.FC<PostFooterProps> = ({ stateStore }) => {
    return (
        <div className="post-footer">
            <div className="labels">{stateStore.visiblePost.tagNames.join(', ')}</div>
        </div>
    );
};

PostFooter.displayName = 'PostFooter';

export default observer(PostFooter);
