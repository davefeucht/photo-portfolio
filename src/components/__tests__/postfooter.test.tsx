import { render } from '@testing-library/react';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';

import PostFooter from '../PostFooter/PostFooter';

test('PostFooter displays', async () => {
    const tagNames = ['something', 'somethingelse'];
    const { container } = render(
        <MemoryRouter>
            <PostFooter tagNames={tagNames} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
});
