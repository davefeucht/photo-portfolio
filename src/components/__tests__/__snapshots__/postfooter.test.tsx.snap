// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFooter displays 1`] = `
<div
  class="post-footer"
>
  <div
    class="labels"
  >
    something, somethingelse
  </div>
</div>
`;
